hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/streaming/RequestResponseMarshaller.java

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.hedera.node.app.blocks.impl.streaming;
+
+import com.hedera.hapi.block.PublishStreamRequest;
+import com.hedera.hapi.block.PublishStreamResponse;
+import com.hedera.pbj.runtime.Codec;
+import com.hedera.pbj.runtime.ParseException;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import io.grpc.MethodDescriptor;
+import io.helidon.grpc.core.MarshallerSupplier;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class RequestResponseMarshaller<T> implements MethodDescriptor.Marshaller<T> {
+    private final Codec<T> codec;
+
+    RequestResponseMarshaller(Class<T> clazz) {
+        if (clazz == PublishStreamRequest.class) {
+            this.codec = (Codec<T>) PublishStreamRequest.PROTOBUF;
+        } else if (clazz == PublishStreamResponse.class) {
```

Comment:
Do these need to be `.equals()` instead of `==`?

Reply:
I think using `==` specifically for class objects should be fine, but I can update it to `.equals()` if required.
Do we have general preference for this one e.g. style guide?